r/pkgdown/extra.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * adapted from rmarkdown/inst/rmd/h/navigation-1.1/tabsets.js
```

Comment:
Why do we have to copy this from rmarkdown? Does `rmarkdown` not make it available? Can we load https://raw.githubusercontent.com/rstudio/rmarkdown/master/inst/rmd/h/navigation-1.1/tabsets.js at page load time in the pkgdown site?

I'm not sure we can do this. [`rmarkdown` is GPL-3 license](https://github.com/rstudio/rmarkdown/blob/master/DESCRIPTION#L113), which is copyleft and on the [forbidden list](https://www.apache.org/legal/resolved.html#category-x). We can of course *use* `rmarkdown`, we just can't copy its code inside of apache/arrow.

Reply:
pkgdown doesn't pull this code in to the javascript it loads, so it doesn't contain it itself (I've thought about sending a PR to add it there, [though they seemed down on the idea generally](https://github.com/r-lib/pkgdown/issues/981))

This code isn't _exactly_ what rmarkdown uses since the pkgdown structure is slightly different than the standard rmarkdown html template.

The [underlying jquery plugin](https://github.com/aidanlister/jquery-stickytabs) is MIT licensed, the `buildTabsets` is the only code that is coming from rmarkdown (the `two $(document.read(...)` functions I wrote as part of dittodb).

Reply:
Ok, I think I've resolved this, we now get that javascript file from GH (though we have to use this JS cdn to get it with the correct mimetype).

Then I managed to work some magic that edits the function like I needed in place. As far as I can see building it locally it works exactly like I expect it should.